lambda_code/src/fhir_controller.py

```diff
@@ -99,30 +116,76 @@ def delete_immunization(self, aws_event):
             return self.create_response(500, unhandled_error.to_operation_outcome())
 
     def search_immunizations(self, aws_event) -> dict:
-        params = aws_event["queryStringParameters"]
-
-        if "nhsNumber" not in params:
-            return self._create_bad_request("Query parameter 'nhsNumber' is mandatory")
-        if "diseaseType" not in params:
-            return self._create_bad_request("Query parameter 'diseaseType' is mandatory")
-
-        result = self.fhir_service.search_immunizations(params["nhsNumber"], params["diseaseType"])
-
+        http_method = aws_event.get("httpMethod")
+        nhs_number_list = []
+        disease_type_list = []
+        nhs_number_param = "-nhsNumber"
+        disease_type_param = "-diseaseType"
+        if http_method == "POST":
+            content_type = aws_event.get("headers", {}).get("Content-Type")
+            if content_type == "application/x-www-form-urlencoded":
+                body = aws_event["body"]
+                decoded_body = base64.b64decode(body).decode("utf-8")
+                parsed_body = parse_qs(decoded_body)
+                nhs_number_list = parsed_body.get(nhs_number_param)
+                disease_type_list = parsed_body.get(disease_type_param)
+                
+        parsed_query_params = aws_event.get("queryStringParameters", {})
+        # AWS API Gateway doesnot support multiple query string parameters with the same name, so it will not be array
```

Comment:
API Gateway does support this on another field. They're under `multiValueQueryStringParameters` and contain all the parameters, even singles, so you don't have to combine it with `queryStringParameters` or anything like that.

https://docs.aws.amazon.com/apigateway/latest/developerguide/set-up-lambda-proxy-integrations.html#api-gateway-simple-proxy-for-lambda-input-format

Reply:
Yeah but we are not using multiValueQueryStringParameters, thats why dont need it. We are passing our params as queryStringParameters

Reply:
We don't choose how the params are passed, API Gateway always puts them in both fields. If we use `queryStringParameters` in our code then we're losing the other parameters so we need to be using `multiValueQueryStringParameters` in order to make the URL params work the same way as the POST content params.

Reply:
If we want to handle multiple params of same name in request, we should create another ticket for it, its out of scope of this ticket.

Reply:
The changes you've made have handled it now so no need.